_data/import-public.rb

```diff
@@ -12,6 +12,23 @@
 require 'hash-joiner'
 require 'safe_yaml'
 
+PROJECT_FILES = Dir["private/projects/*.yml"]
+
+private_projects_yaml_file = "private/projects.yml"
+project_data = SafeYAML.load_file(private_projects_yaml_file, :safe=>true)
+PROJECT_FILES.each do |source|
+  data = SafeYAML.load_file(source, :safe=>true)
+  unless data
+    puts "Failed to parse #{source}"
+    exit 1
+  end
+  data = data.first  # one project per yaml file in projects dir
+  index = project_data.index { |proj| proj['name'] == data['name'] }
+  index = project_data.length  if index.nil?
+  project_data[index] = data
+end
+open(private_projects_yaml_file, 'w') {|outfile| outfile.puts project_data.to_yaml}
```

Comment:
Ah, just realized there's something else you can do with `hash-joiner` here:

```ruby
private_projects_yaml_file = "private/projects.yml"
project_data = SafeYAML.load_file(private_projects_yaml_file, :safe=>true)
project_files_data = []
PROJECT_FILES.each do |source|
  data = SafeYAML.load_file(source, :safe=>true)
  unless data
    puts "Failed to parse #{source}"
    exit 1
  end
  project_files_data << data.first  # one project per yaml file in projects dir
end
HashJoiner.join_array_data 'name', project_data, project_files_data
open(private_projects_yaml_file, 'w') {|outfile| outfile.puts project_data.to_yaml}
```
Give it a spin? @gboone The `filter-yaml-files` idea I had in mind pertains to the `YAML_FILES` loop after this one. :-)